tasks/cli.yaml

```diff
@@ -0,0 +1,37 @@
+---
+- name: validate connection is network_cli
+  fail:
+    msg: "expected connection network_cli, got {{ ansible_connection }}"
+  when: ansible_connection != 'network_cli'
+
+- name: check for required fact ansible_network_os
+  fail:
+    msg: "missing required fact: ansible_network_os"
+  when: ansible_network_os is undefined
+
+- name: validate the engine variable
+  fail:
+    msg: "invalid option for engine"
+  when: engine is defined and engine not in ['command_parser', 'textfsm_parser', 'text_parser', 'textfsm']
+
+- name: check for required fact command
+  fail:
+    msg: "missing required fact: command"
+  when: command is undefined
+
+- name: check for required fact parser for engine
+  fail:
+    msg: "missing required fact: parser"
+  when: engine is defined and parser is undefined
+
+- name: run command on remote network node
+  cli:
+    command: "{{ command }}"
+  when: parser is undefined
+
+- name: run command on remote network node and use engine to parse output into JSON facts
+  cli:
+    command: "{{ command }}"
+    parser: "{{ parser }}"
+    engine: "{{ engine | default('command_parser') }}"
```

Comment:
Set the default value to omit so as to not override the default provided by the module

`{{ engine | default(omit) }}"`